pkgs/development/web/deno/default.nix

```diff
@@ -0,0 +1,103 @@
+{ stdenv, fetchurl, fetchFromGitHub, rust, rustPlatform, python27, installShellFiles }:
+
+let
+  pname   = "deno";
+  version = "1.0.0-rc3";
+
+  denoSrc = fetchFromGitHub {
+    owner  = "denoland";
+    repo   = pname;
+    rev    = "v${version}";
+    sha256 = "0qqsnbndb0p20w1rnglpv03b9l6iwm6vbwxzikd7fl2kvzxz5z3m";
+
+    fetchSubmodules = true;
+  };
+  cargoSha256 = "0nvvhll55lgni231nb3fg4dff7pb35qybfdcaphr1sa5rfbmmc7h";
+
+  rustyV8Ver = "0.4.2";
+  rustyV8Lib = fetchlib "rusty_v8" "${rustyV8Ver}" {
+    x86_64-linux  = "1ac6kv3kv087df6kdgfd7kbh24187cg9z7xhbz6rw6jjv4ci2zbi";
+    aarch64-linux = "06iyjx4p4vp2i81wdy0vxai2k18pki972ff7k0scjqrgmnav1p8k";
+    x86_64-darwin = "02hwbpsqdzb9mvfndgykvv44f1jig3w3a26l0h26hs5shsrp47jv";
+  };
+
+  arch = rust.toRustTarget stdenv.hostPlatform;
+  fetchlib = name: version: sha256: fetchurl {
+    url    = "https://github.com/denoland/${name}/releases/download/v${version}/librusty_v8_release_${arch}.a";
+    sha256 = sha256."${stdenv.hostPlatform.system}";
+  };
+in
+rustPlatform.buildRustPackage rec {
+  inherit pname version cargoSha256;
+
+  src = denoSrc;
+
+  nativeBuildInputs = [
+    python27          # rusty_v8's download script requires python2.7
+    installShellFiles # Install completions post-install
+  ];
+
+  preBuild = ''
+    # Check the rusty_v8 lib downloaded matches the Cargo.lock file
+    rusty_v8_ver="$(grep 'name = "rusty_v8"' -A 1 Cargo.lock | grep "version =" | cut -d\" -f2)"
+    if [ "$rusty_v8_ver" != "${rustyV8Ver}" ]; then
+      printf "%s\n%s\n%s\n" 2>&1 \
```

Comment:
Nit: you mean `1>&2` (or `>&2`).

Reply:
Nice spot

Reply:
I've moved it to the following:

```sh
printf "%s\n" >&2 \
  "version mismatch between 'rusty_v8' in Cargo.lock and downloaded library:" \
  "  wanted: ${rustyV8Lib.meta.version}" \
  "  got:    $rusty_v8_ver"
```

My wanted and got were the wrong way around :sweat: 
Now pulls version directly from the lib (added `meta = { inherit version; };` to it's `fetchurl`)
The repeating `%s\n`s were superfluous.
The error msg and format are now more consistent with some of the other mismatch logs

```ini
# fetchgit example output:
hash mismatch in fixed-output derivation '/nix/store/mm332zdbgmmlvf14522af59nzidnry20-source':
  wanted: sha256:0qqsnbndb0p20w1rnglpv03b9l6iwm6vbwxzikd7fl2kvzxz5z3m
  got:    sha256:0k8mqy1hf9hkp60jhd0x4z814y36g51083b3r7prc69ih2523hd1

# previous output:
rusty_v8 version in Cargo.lock doesn't match downloaded release
Wanted: 0.4.2
Got:    0.4.1

# new output:
version mismatch between 'rusty_v8' in Cargo.lock and downloaded library:
  wanted: 0.4.1
  got:    0.4.2
```